qiskit/compiler/transpile.py

```diff
@@ -336,14 +374,21 @@ def _parse_transpile_args(circuits, backend,
     initial_layout = _parse_initial_layout(initial_layout, circuits)
     layout_method = _parse_layout_method(layout_method, num_circuits)
     routing_method = _parse_routing_method(routing_method, num_circuits)
+    scheduling_method = _parse_scheduling_method(scheduling_method, num_circuits)
+    durations = None
+    if scheduling_method:
+        from qiskit.transpiler.instruction_durations import InstructionDurations
+        durations = InstructionDurations.from_backend(backend).update(instruction_durations)
```

Comment:
I'm pretty sure `InstructionDuration` and `DurationMapper` can be largely simplified. When I write my own code, I write this:
```
backend = ...
def duration(gate, qubits, backend=backend):
    return backend.properties().gate_length(gate, qubits) // backend.configuration().dt
```
The weakness here is that 1) There's no memoization and 2) I don't want to have to pass backend.

I think we probably need something in between. Something that doesn't suffer from these weaknesses ^^^ but isn't as heavy handed as this `InstructionDuration` that requires two classes

Reply:
Maybe we can define a method `BaseBackend.duration` which can memoize the instruction durations. It doesn't fit the defaults/properties/config model, but it would work. Then I could simply write:
```
backend = ....
backend.duration('u3', 0)
# or alias duration = backend.duration
```

Reply:
I put all the functionality related to duration into `InstructionDuration` class in this PR. I think we need such a class to manage *manual* `instruction_durations`. I like your idea that backend should have more user-friendly interface to get duration of an instruction. It will be very useful for majority of users who use *default* instruction durations of a backend. And it will simplify the implementation of `InstructionDuration`. I think we should address the issue in a follow-up PR.

Reply:
Ok, would you mind creating the issue now and sharing here?

Reply:
https://github.com/Qiskit/qiskit-terra/issues/4703